geoapps/utils/surveys.py

```diff
@@ -377,3 +377,37 @@ def split_dcip_survey(
             line_surveys.append(line_survey)
 
     return line_surveys
+
+
+def get_unique_locations(survey: BaseSurvey) -> np.ndarray:
+    if survey.source_list:
+        locations = []
+        for source in survey.source_list:
+            source_location = source.location
```

Comment:
could probably just use source.location everywhere, no?

Reply:
Need to test for the type as sometimes it's a list, sometimes just an array.